### PR TITLE
ingresscontroller pkg added

### PR DESCRIPTION
**Pull request by elenagerman**
ingresscontroller pkg added

### CODE REVIEW

File: pkg/ingresscontroller/ingresscontroller.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ingresscontrollerName      = "default"
+	ingresscontrollerNamespase = "openshift-ingress-operator"
+)
+
+// Builder provides a struct for ingresscontroller object from the cluster and a ingresscontroller definition.
```
Comment (achuzhoy) [inline pkg/ingresscontroller/ingresscontroller.go:21]: nit: and an ingresscontroller

File: pkg/ingresscontroller/ingresscontroller.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	return &builder, nil
+}
+
+// Get fetches existing ingresscontroller from cluster.
+func (builder *Builder) Get() (*v1.IngressController, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Pulling existing ingresscontroller with name %s under namespace %s from cluster",
```
Comment (achuzhoy) [inline pkg/ingresscontroller/ingresscontroller.go:62]: Fetching

**Comment by achuzhoy** [comment]
Looks good to me apart from some minor comments

File: pkg/ingresscontroller/ingresscontroller.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "github.com/openshift/api/operator/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ingresscontrollerName      = "default"
+	ingresscontrollerNamespase = "openshift-ingress-operator"
```
Comment (kononovn) [inline pkg/ingresscontroller/ingresscontroller.go:18]: Please remove this constant. It should be parameter. You can have this constant in test code

File: pkg/ingresscontroller/ingresscontroller.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "github.com/openshift/api/operator/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ingresscontrollerName      = "default"
```
Comment (kononovn) [inline pkg/ingresscontroller/ingresscontroller.go:17]: Please remove this constant. It should be parameter. You can have this constant in test code

File: pkg/ingresscontroller/ingresscontroller.go [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+	// ingresscontroller definition, used to create the ingresscontroller object.
+	Definition *v1.IngressController
+	// Created ingresscontroller object.
+	Object *v1.IngressController
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// Pull loads an existing ingresscontroller into Builder struct.
+func Pull(apiClient *clients.Settings) (*Builder, error) {
```
Comment (kononovn) [inline pkg/ingresscontroller/ingresscontroller.go:32]: Please add name and nsname to this function

File: pkg/ingresscontroller/ingresscontroller.go [hunk 6]
```diff
@@ -0,0 +1,139 @@
+package ingresscontroller
```
Comment (kononovn) [inline pkg/ingresscontroller/ingresscontroller.go:1]: Please rename package to ingress and directory ./pkg/ingress. Under ingress we could have several resources like:
1. Ingress
2. Ingresscontroller 
and more

File: pkg/ingresscontroller/ingresscontroller.go [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+	}
+
+	builder.Definition.CreationTimestamp = metaV1.Time{}
+	builder.Definition.ResourceVersion = ""
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+
+	return builder, err
+}
+
```
Comment (kononovn) [inline pkg/ingresscontroller/ingresscontroller.go:114]: Please add Delete function.

**Review by kononovn — CHANGES_REQUESTED** [review]
Overall looks good, just few comments

File: pkg/ingress/ingresscontroller.go [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Update renovates a Builder in the cluster and stores the created object in struct.
+func (builder *Builder) Update() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating the Builder %s in namespace %s",
```
Comment (kononovn) [inline pkg/ingress/ingresscontroller.go:92]: Please change log to "Updating the ingresscontroller %s in namespace...."

**Review by kononovn — APPROVED** [review]
apart 2 comments above **lgtm**

**Comment by kononovn** [comment]
@elenagerman Please fix 2 minor comments above and I'll merge this PR

**Pull request merged**